lua/neogit/lib/git/status.lua

```diff
@@ -15,6 +15,11 @@ local function update_file(file, mode, name)
   return setmetatable({ mode = mode, name = name, has_diff = has_diff, diff = diff }, mt or {})
 end
 
+-- Generic pattern for matching tag ref and distance from rev
+-- Unfortunately lua's pattern matching isn't that complete so
+-- some cases may be dropped.
+local tag_pattern = "(.-)%-([0-9]+)%-g[0-9a-z]+$"
```

Comment:
```suggestion
local tag_pattern = "(.-)%-([0-9]+)%-g%x+$"
```
`%x` matches hexidecimal chars :)
https://www.lua.org/pil/20.2.html